src/commands/run.rs

```diff
@@ -142,13 +142,24 @@ impl RunCommand {
         {
             Ok(()) => (),
             Err(e) => {
-                // If the program exited because of a trap, return an error code
-                // to the outside environment indicating a more severe problem
-                // than a simple failure.
-                if e.is::<Trap>() {
+                // If the program exited because of a non-zero exit status, print
+                // a message and exit.
+                if let Some(trap) = e.downcast_ref::<Trap>() {
                     // Print the error message in the usual way.
                     eprintln!("Error: {:?}", e);
 
+                    if let TrapReason::I32Exit(status) = trap.reason() {
+                        // On Windows, exit status 3 indicates an abort (see below),
+                        // so just return 1 indicating a non-zero status.
```

Comment:
This seems a bit odd to me, why not expose exiting with all sorts of error codes on Windows?

Reply:
(or put another way I don't fully understand the comment as to why Windows is special here)

Reply:
I don't have a strong sense of what Windows wants here; I just didn't want to overload exit status 3 as meaning both "program trapped" and "program explicitly exited with status 3".

Am I reading too much into the docs by assuming that using 3 for an abort is a meaningful convention?